dd-sdk-android/src/main/kotlin/com/datadog/android/rum/tracking/AndroidTvComponentPredicate.kt

```diff
@@ -0,0 +1,47 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.rum.tracking
+
+import androidx.fragment.app.Fragment
+
+/**
+ * Special ComponentPredicate which can be used when defining tracking strategy for
+ * Android TV applications. By using this predicate, by default the extra
+ * view events created by `android.leanback` package helper fragments are excluded:
+ * `androidx.leanback.app.HeadersSupportFragment`, `androidx.leanback.app.RowsSupportFragment`.
+ * In case you also want to use your own implementation of a ComponentPredicate you can wrap
+ * it inside the `AndroidTvComponentPredicate`:
+ * ```
+ * val predicate = AndroidTvComponentPredicate(yourOwnPredicate)
+ * ```
+ */
+class AndroidTvComponentPredicate(
+    internal val wrappedPredicate: ComponentPredicate<Fragment>
+) : ComponentPredicate<Fragment> by wrappedPredicate {
+
+    constructor() : this(AcceptAllSupportFragments())
+
+    override fun accept(component: Fragment): Boolean {
+        if (isLeanbackApiHelperFragment(component)) return false
+        return wrappedPredicate.accept(component)
+    }
+
+    private fun isLeanbackApiHelperFragment(fragment: Fragment): Boolean {
+        return fragment.javaClass.name in LEANBACK_FRAGMENTS
+    }
+
+    companion object {
+        private const val LEANBACK_HEADERS_SUPPORT_FRAGMENT_NAME =
+            "androidx.leanback.app.HeadersSupportFragment"
+        private val LEANBACK_ROWS_SUPPORT_FRAGMENT_NAME =
+            "androidx.leanback.app.RowsSupportFragment"
+        private val LEANBACK_FRAGMENTS = setOf(
```

Comment:
Are those the only Fragments in the leanback/android tv libraries that could be used that we'd want to exclude?
Also are those fragments created by the Android TV library or are those something the app developper instantiate themselves?

Reply:
These are the ones I could find so far in the sample app I built. There can be more and they are created automatically by `leanback` API under the hood.

Reply:
Can we please take time to find what other automatic components could be added by the leanback library?